File: backend-rekomendasi-resep-makanan/app/Models/Recipe.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Support\Facades\Storage;

class Recipe extends Model
{
    use HasFactory;

    /**
     * The attributes that are mass assignable.
     *
     * @var array<int, string>
     */
    protected $fillable = [
        'nama_resep',
        'category_id',
        'deskripsi',
        'bahan',
        'langkah_langkah',
        'image', // Tambahkan 'image'
    ];

    /**
     * The attributes that should be cast.
     *
     * @var array<string, string>
     */
    protected $casts = [
        'bahan' => 'array',
        'langkah_langkah' => 'array',
    ];

    /**
     * Append an attribute to the model's JSON form.
     *
     * @var array
     */
    protected $appends = ['image_url'];

    /**
     * Get the full URL forthe recipe image.
     *
     * @return string|null
     */
    public function getImageUrlAttribute()
    {
        // Jika image null atau string kosong, return null
        if (empty($this->image) || trim($this->image) === '') {
            return null; 
        }

        // Jika sudah full URL (misal dummy data), kembalikan langsung
        if (str_starts_with($this->image, 'http')) {
            return $this->image;
        }

        // Jika path relatif, tambahkan base URL storage
        return asset('storage/' . $this->image);
    }

    public function category()
    {
        return $this->belongsTo(Category::class);
    }
}

File: Rekomendasi Resep Makanan/app/src/main/java/com/rekomendasiresepmakanan/data/model/Recipe.kt
package com.rekomendasiresepmakanan.data.model

import androidx.compose.runtime.Immutable
import com.google.gson.annotations.SerializedName

@Immutable
data class Recipe(
    val id: Int,
    @SerializedName("nama_resep")
    val title: String,
    @SerializedName("category_id")
    val categoryId: Int,
    val category: Category?,
    @SerializedName("deskripsi")
    val description: String,
    @SerializedName("bahan")
    val ingredients: List<String>,
    @SerializedName("langkah_langkah")
    val steps: List<String>,
    @SerializedName("gambar")
    val image: String,
    val imageRes: Int? = null
)

